Factory/DiactorosFactory.php

```diff
@@ -65,6 +65,7 @@ public function createRequest(Request $symfonyRequest)
             ->withCookieParams($symfonyRequest->cookies->all())
             ->withQueryParams($symfonyRequest->query->all())
             ->withParsedBody($symfonyRequest->request->all())
+            ->withRequestTarget($symfonyRequest->getRequestUri())
```

Comment:
Yep I understand what you're saying, I was thinking the same myself. Yet when I looked at the implementation I thought it best to be explicit here. This implementation tokenizes the URL, and pieces together a request target if it is not set explicitly. By providing the request target string, you're giving the implementation the correct data - unfortunately the constructor is not giving you that opportunity.  https://github.com/zendframework/zend-diactoros/blob/master/src/RequestTrait.php#L124-L129